test/e2e/network_test.go

```diff
@@ -140,6 +141,101 @@ func testEastWestNetworkingAcrossWindowsNodes(t *testing.T) {
 	}
 }
 
+// testNorthSouthNetworking deploys a Windows Server pod, and tests that we can network with it from outside the cluster
+func testNorthSouthNetworking(t *testing.T) {
+	testCtx, err := NewTestContext(t)
+	require.NoError(t, err)
+
+	for _, node := range gc.nodes {
+		affinity, err := getAffinityForNode(&node)
+		require.NoError(t, err, "Could not get affinity for node")
+
+		// Deploy a webserver pod on the new node. This is prone to timing out due to having to pull the Windows image
+		// So trying multiple times
+		var winServerDeployment *appsv1.Deployment
+		for i := 0; i < deploymentRetries; i++ {
+			winServerDeployment, err = testCtx.deployWindowsWebServer("win-webserver-"+
+				strings.ToLower(node.Status.NodeInfo.MachineID), affinity)
+			if err == nil {
+				break
+			}
+		}
+		require.NoError(t, err, "could not create Windows Server deployment")
+		defer testCtx.deleteDeployment(winServerDeployment.Name)
+
+		// Create a load balancer svc to expose the webserver
+		loadBalancer, err := testCtx.createLoadBalancer("win-webserver-"+strings.ToLower(node.Status.NodeInfo.MachineID),
+			*winServerDeployment.Spec.Selector)
+		require.NoError(t, err, "Could not create load balancer for Windows Server")
+		defer testCtx.deleteService(loadBalancer.Name)
+		loadBalancer, err = testCtx.waitForLoadBalancerIngress(loadBalancer.Name)
+		require.NoError(t, err, "Error waiting for load balancer ingress")
+
+		// Try and read from the webserver through the load balancer. The load balancer takes a fair amount of time,
+		// ~3 min, to start properly routing connections.
+		resp, err := retryGET("http://" + loadBalancer.Status.LoadBalancer.Ingress[0].Hostname)
+		require.NoError(t, err, "Could not GET from load balancer: %v", loadBalancer)
+		assert.Equal(t, http.StatusOK, resp.StatusCode, "Non 200 response from webserver")
+		resp.Body.Close()
```

Comment:
This can become a helper or a series of helpers that clearly allows the reader to see the high level function the loop is doing.